fearless/Modules/BackupPassword/BackupPasswordPresenter.swift

```diff
@@ -0,0 +1,134 @@
+import Foundation
+import SoraFoundation
+import SSFCloudStorage
+
+protocol BackupPasswordViewInput: ControllerBackedProtocol, HiddableBarWhenPushed, LoadableViewProtocol {
+    func didReceive(walletName: String)
+    func setPasswordInputViewModel(_ viewModel: InputViewModelProtocol)
+}
+
+protocol BackupPasswordInteractorInput: AccountImportInteractorInputProtocol {
+    func setup(with output: BackupPasswordInteractorOutput)
+    func importBackup(account: OpenBackupAccount, password: String)
+}
+
+final class BackupPasswordPresenter {
+    // MARK: Private properties
+
+    private weak var view: BackupPasswordViewInput?
+    private let router: BackupPasswordRouterInput
+    private let interactor: BackupPasswordInteractorInput
+
+    private let backupAccounts: [BackupAccount]
+
+    private let passwordInputViewModel = {
+        InputViewModel(inputHandler: InputHandler(predicate: NSPredicate.notEmpty))
+    }()
+
+    // MARK: - Constructors
+
+    init(
+        backupAccounts: [BackupAccount],
+        interactor: BackupPasswordInteractorInput,
+        router: BackupPasswordRouterInput,
+        localizationManager: LocalizationManagerProtocol
+    ) {
+        self.backupAccounts = backupAccounts
+        self.interactor = interactor
+        self.router = router
+        self.localizationManager = localizationManager
+    }
+
+    // MARK: - Private methods
+
+    private func provideViewModel() {
+        guard let account = backupAccounts.first(where: { $0.current })?.account else {
+            return
+        }
+        let walletName = account.name ?? account.address
+        view?.didReceive(walletName: walletName)
+    }
+
+    private func proceed(with backup: OpenBackupAccount) {
+        guard
+            let passphrase = backup.passphrase,
+            let mnemonic = interactor.createMnemonicFromString(passphrase),
+            let substrateDerivationPath = backup.substrateDerivationPath,
+            let ethDerivationPath = backup.ethDerivationPath,
+            let backupCryptoType = backup.cryptoType,
+            let rawValue = UInt8(backupCryptoType),
+            let cryptoType = CryptoType(rawValue: rawValue),
+            let name = backup.name
+        else {
+            return
```

Comment:
alert?

Reply:
Have you idea for text of message? Need fetch approve and will insert in localise